dqc_us_rules/dqc_us_0001.py

```diff
@@ -0,0 +1,323 @@
+    """
+    Check extension members under the given axis.
+
+    :param axis: The axis to check
+    :type axis: :class:'~arelle.ModelDTSObject.ModelConcept'
+    :param axis_config: The axis-specific config.
+    :type axis_config: dict
+    :param relset: The relationshipSet for the axis.
+    :type relset: set
+    :param val: val from which to gather end dates
+    :type val: :class:'~arelle.ModelXbrl.ModelXbrl'
+    :param role: The role for the relationship
+    :type role: str
+    :return: No direct return
+    :rtype: None
+    """
+    allow_all = (len(axis_config[_EXTENSIONS_KEY]) > 0 and
+                 axis_config[_EXTENSIONS_KEY][0] == '*')
+    if not allow_all:
+        allowed_extensions = axis_config[_EXTENSIONS_KEY]
+        for child in _all_members_under(axis, relset):
+            if _is_extension(child, val):
+                if child.qname.localName not in allowed_extensions:
+                    fact = facts.axis_member_fact(
+                        axis.qname.localName,
+                        child.qname.localName,
+                        val.modelXbrl
+                    )
+                    if fact is not None:
+                        val.modelXbrl.error(
+                            '{base_key}.{extension_key}'.format(
+                                base_key=_CODE_NAME,
+                                extension_key=axis_config[_RULE_INDEX_KEY]
+                            ),
+                            messages.get_message(_CODE_NAME, _EXT_FACT_KEY),
+                            axis=axis.label(),
+                            member=child.label(),
+                            modelObject=fact,
+                            ruleVersion=_RULE_VERSION
+                        )
+                    else:
+                        val.modelXbrl.error(
+                            '{base_key}.{extension_key}'.format(
+                                base_key=_CODE_NAME,
+                                extension_key=axis_config[_RULE_INDEX_KEY]
+                            ),
+                            messages.get_message(_CODE_NAME, _NO_FACT_KEY),
+                            axis=axis.label(),
+                            member=child.label(),
+                            group=role,
+                            ruleVersion=_RULE_VERSION
+                        )
+
+
+def _is_extension(concept, val):
+    return (concept.qname.namespaceURI not in
+            val.disclosureSystem.standardTaxonomiesDict)
+
+
+def _is_concept(concept):
+    """
+    This utility method should be used instead of None checks on
+    arc.fromModelObject or arc.toModelObject.
+
+    :param concept: The concept to check
+    :type concept: :class:'~arelle.ModelDTSObject.ModelConcept'
+    :return: True if it's a valid concept.  False if not.
+    """
+    return (concept is not None and
+            isinstance(concept, ModelConcept) and
+            concept.qname is not None)
```

Comment:
It's a direct wf_arelle port, so yeah, I totally agree that it can be simplified.

Not sure if I understand the idea of defaulting to None in the function definition.  A plain old instance and qname check should work, I think, as you suggested.

As for the qname check - doesn't python always complain about implicit none checks via a futurewarning?